doc/source/networking.rst

```diff
@@ -205,6 +208,33 @@ To configure both IPv4 and IPv6:
 
 .. note:: For VNET jails, a default route has to also be specified.
 
+Configuring a VNET Jail with Netgraph Interfaces
+++++++++++++++++++++++++++++++++++++++++++++++++
+
+:manpage:`netgraph(4)` provides a flexible, modular system for connecting
+network interfaces within the FreeBSD kernel. :command:`iocage` will create
+:manpage:`ng_eiface(4)` interfaces for a jail and attach them to a
+:manpage:`ng_bridge(4)` if the ``netgraph`` parameter is set on the jail:
+
+:samp:`# iocage set netgraph="on" examplejail`
+
+All of the *VNET* instructions apply to Netgraph-enabled jails, but jails
+attached to a :manpage:`ng_bridge(4)` can be placed on a host-only, routed
+network so that the host's firewall can monitor and filter traffic. At the
+present time, the host's :manpage:`ng_eiface(4)` must be manually created
```

Comment:
What's the blocker for making this automatic like the rest of the networking in iocage? I think a user may find it odd that everything but netgraph works automatically within iocage's interface.

Reply:
I already have the code that sets up the host's ng_eiface, BUT it's more of an issue of what tasks really belong to iocage. In the case of an if_bridge, iocage creates the bridge and adds the host interface, but it does not create or configure the host interface. With ng_eiface it's tempting to have iocage create and configure the interface because, frankly, it's a pain in the neck to do via the rc scripts right now. I have a patch to `network.subr` that allows me to specify `netgraph_interfaces` in `rc.conf`, which are created during startup and then can be configured in the normal `rc.conf` way.

The other issue is that the host's interface configuration doesn't really belong to any one jail. We can easily check for and create an `ng_eiface` node for the host if it doesn't exist already, but when would it get an IP address? Where is the authoritative configuration for that interface? Right now, `rc.conf` seems like the only logical choice unless iocage got some notion of "global" or "host" configuration values. For both those reasons, I left any kind of host interface configuration out.

Reply:
Agreed, it's a conundrum as is.